pypit/core/artracewave.py

```diff
@@ -0,0 +1,1235 @@
+            arpca.pca_plot(slf, outpar, ofit, 'Arc', pcadesc=pcadesc, addOne=False)
+        # Extrapolate the remaining orders requested
+        orders = 1.0 + np.arange(norders)
+        extrap_tilt, outpar = arpca.extrapolate(outpar, orders, function=settings_argflag['trace']['slits']['function'])
+        tilts = extrap_tilt
+#        arqa.pca_arctilt(slf, tiltang, centval, tilts)
+        arpca.pca_arctilt(slf, tiltang, centval, tilts)
+    else:
+        outpar = None
+        msgs.warn("Could not perform a PCA when tracing the order tilts" + msgs.newline() +
+                  "Not enough well-traced orders")
+        msgs.info("Attempting to fit tilts by assuming the tilt is order-independent")
+        xtiltfit = np.array([])
+        ytiltfit = np.array([])
+        for o in range(tiltang.shape[1]):
+            w = np.where(tiltang[:, o] != -999999.9)
+            if np.size(w[0]) != 0:
+                xtiltfit = np.append(xtiltfit, centval[:, o][w])
+                ytiltfit = np.append(ytiltfit, tiltang[:, o][w])
+        if np.size(xtiltfit) > settings_argflag['trace']['slits']['tilts']['disporder'] + 2:
+            tcoeff = arutils.func_fit(xtiltfit, ytiltfit, settings_argflag['trace']['slits']['function'],
+                                      settings_argflag['trace']['slits']['tilts']['disporder'],
+                                      minv=0.0, maxv=msarc.shape[0] - 1)
+            tiltval = arutils.func_val(tcoeff, xv, settings_argflag['trace']['slits']['function'], minv=0.0,
+                                       maxv=msarc.shape[0] - 1)
+            tilts = tiltval[:, np.newaxis].repeat(tiltang.shape[1], axis=1)
+        else:
+            msgs.warn("There are still not enough detections to obtain a reliable tilt trace")
+            msgs.info("Assuming there is no tilt")
+            tilts = np.zeros_like(slf._lordloc)
+
+    # Generate tilts image
+#    print('calling tilts_image')
+#    t = time.clock()
+#    _tiltsimg = arcytrace.tilts_image(tilts, slf._lordloc[det-1], slf._rordloc[det-1],
+#                                     settings.argflag['trace']['slits']['pad'], msarc.shape[1])
+#    print('Old tilts_image: {0} seconds'.format(time.clock() - t))
+#    t = time.clock()
+    tiltsimg = new_tilts_image(tilts, slf._lordloc[det-1], slf._rordloc[det-1],
+                                settings_argflag['trace']['slits']['pad'], msarc.shape[1])
+#    print('New tilts_image: {0} seconds'.format(time.clock() - t))
+#    assert np.sum(_tiltsimg != tiltsimg) == 0, 'Difference between old and new tilts_image'
+
+    return tiltsimg, satmask, outpar
+
+
+def multislit_tilt(msarc, lordloc, rordloc, pixlocn, pixcen, slitpix, det,
+                   maskslits, tilt_settings, settings_spect, setup,
+                   maskval=-999999.9, doqa=False, wv_calib=None):
+    """ Determine the spectral tilt of each slit in a multislit image
+
+    Parameters
+    ----------
+    slf : Class instance
```

Comment:
docstring to be updated

Reply:
I have now commented this code out.  It is superfluous.